streams/src/test/java/org/apache/kafka/streams/kstream/internals/KStreamImplTest.java

```diff
@@ -2330,4 +2331,484 @@ public void shouldNotAllowNullNamedOnProcessWithStores() {
+
+        assertThat(
+            topologyDescription,
+            equalTo("Topologies:\n" +
+                "   Sub-topology: 0\n" +
+                "    Source: KSTREAM-SOURCE-0000000000 (topics: [input])\n" +
+                "      --> KSTREAM-MAP-0000000001\n" +
+                "    Processor: KSTREAM-MAP-0000000001 (stores: [])\n" +
+                "      --> KSTREAM-FILTER-0000000005\n" +
+                "      <-- KSTREAM-SOURCE-0000000000\n" +
+                "    Processor: KSTREAM-FILTER-0000000005 (stores: [])\n" +
+                "      --> KSTREAM-SINK-0000000004\n" +
+                "      <-- KSTREAM-MAP-0000000001\n" +
+                "    Sink: KSTREAM-SINK-0000000004 (topic: KSTREAM-TOTABLE-0000000003-repartition)\n" +
+                "      <-- KSTREAM-FILTER-0000000005\n" +
+                "\n" +
+                "  Sub-topology: 1\n" +
+                "    Source: KSTREAM-SOURCE-0000000011 (topics: [KSTREAM-KEY-SELECT-0000000008-repartition])\n" +
+                "      --> KSTREAM-JOIN-0000000012\n" +
+                "    Processor: KSTREAM-JOIN-0000000012 (stores: [KSTREAM-TOTABLE-STATE-STORE-0000000002])\n" +
+                "      --> KSTREAM-AGGREGATE-0000000014\n" +
+                "      <-- KSTREAM-SOURCE-0000000011\n" +
+                "    Source: KSTREAM-SOURCE-0000000006 (topics: [KSTREAM-TOTABLE-0000000003-repartition])\n" +
+                "      --> KSTREAM-TOTABLE-0000000003\n" +
+                "    Processor: KSTREAM-AGGREGATE-0000000014 (stores: [KSTREAM-AGGREGATE-STATE-STORE-0000000013])\n" +
+                "      --> none\n" +
+                "      <-- KSTREAM-JOIN-0000000012\n" +
+                "    Processor: KSTREAM-TOTABLE-0000000003 (stores: [KSTREAM-TOTABLE-STATE-STORE-0000000002])\n" +
+                "      --> none\n" +
+                "      <-- KSTREAM-SOURCE-0000000006\n" +
+                "\n" +
+                "  Sub-topology: 2\n" +
+                "    Source: KSTREAM-SOURCE-0000000007 (topics: [topic])\n" +
+                "      --> KSTREAM-KEY-SELECT-0000000008\n" +
+                "    Processor: KSTREAM-KEY-SELECT-0000000008 (stores: [])\n" +
+                "      --> KSTREAM-FILTER-0000000010\n" +
+                "      <-- KSTREAM-SOURCE-0000000007\n" +
+                "    Processor: KSTREAM-FILTER-0000000010 (stores: [])\n" +
+                "      --> KSTREAM-SINK-0000000009\n" +
+                "      <-- KSTREAM-KEY-SELECT-0000000008\n" +
+                "    Sink: KSTREAM-SINK-0000000009 (topic: KSTREAM-KEY-SELECT-0000000008-repartition)\n" +
+                "      <-- KSTREAM-FILTER-0000000010\n\n"));
+
+
+        final Matcher matcher = repartitionTopicPattern.matcher(topologyDescription);
+        assertTrue(matcher.find());
+        final String match = matcher.group();
+        assertThat(match, notNullValue());
+        assertTrue(match.endsWith("repartition"));
```

Comment:
This last checks seem redundant to above `assertThat(topologyDescription, equalTo("..."))`?